lib/hacluster.pm

```diff
@@ -1137,4 +1138,47 @@ sub check_iscsi_failure {
     }
 }
 
+=head3 cluster_check_status_by_regex
+
+Remotely check 'crm status' output against regular expression
+=over 3
+
+=item B<SHOW_CLUSTER_STATUS> - Output from 'crm status' command
+
+=item B<REGULAR_EXPRESSION> - Regular expression to match the text to find
+
+=back
+=cut
+
+sub cluster_check_status_by_regex {
+    my ($show_cluster_status, $regular_expression) = @_;     
+    croak 'No crm status output' if (!$show_cluster_status);
+    croak 'No regular expression provided' unless (ref $show_cluster_status eq 'Regexp');
+  
+    my $done = 0;   
+    
+    for my $line (split("\n", $show_cluster_status)) {
+        if ($line =~ $regular_expression) {
+            my @cluster_status = ("Stopped", "Failed", "Pending", "Blocked");
+
+            foreach my $cluster_state (@cluster_status) {
+                if ($line =~ $cluster_state) {
+                    $done = 1;
+                    print "$cluster_state matches the text '$line'\n";
```

Comment:
I'm not sure `print` is usable in this context

Reply:
Yes. better use `record_info()` to show a detail box in the test results, or use `diag()` to print the message in the autoinst.log

Reply:
refactored